### ISSUE TITLE
Movable columns don't snap in in IE

### ISSUE THREAD

**Issue by fortrax** [olifolkerd/tabulator#504]
First let me tell you that I am really happy that I found your project! There are so many tab controls out in the web but your project is an intelligent and elegant selection of what's important and what not.

I have a little issue when using IE11 on a 3.2 sample which is to my mind essentially identical to your "Persistent Column Layout" example:

`           $("#overviewtable").tabulator({
                movableColumns: true,
                columns: [
                    { title: "Name", field: "name", width: 200 },
                    { title: "Date Of Birth", field: "dob", sorter: "date" },
                ],
            });
`

I can click on column headers then, and move them around but when moving over other columns, 
it seems, the "mousemove" event does not get through the placeholderElement (in fact, the debugger in config.mousemove is not hit.) On mouseup, the column snaps back to its original position. Chrome works fine. And IE11 works fine on your original "Persistent Column Layout" example. This brought me to the assumption that I included the wrong CSS or jquery (UI) and tried to approach your web page (you're using a really old jquery, don't you?) but to no success. Any idea? Thank you for reading.

**Comment by olifolkerd** [comment]
Hey,

Thanks for your kind words, it is always great to hear that Tabulator is appreciated,

That would suggest that you have some other CSS or JavaScript inadvertently acting on the Tabulator library, the ***movableColumns*** functionality is fairly complex so one bit of CSS interacting incorrectly could easily break this.

I would suggest that you try creating your tabulator on a plain HTML boiler plate page and reintroduce elements until you can isolate where the interference is coming from.

Thanks for pointing out the old version of jQuery running on the website I have just updated this for the 3.2 example pages. 

Just so you are aware, there is actually very little jQuery and almost no jQuery UI (it only uses the wrapper functionality) used in tabulator as of version 3.0, it is mainly in there to aid previous users.

As of version 4.0 Tabulator will be dependency free with no jQuery needed at all, though it will include a jQuery wrapper for those that still want to use it with jQuery.

Which files are you including in your project at the momnent?

Cheers

Oli

**Comment by fortrax** [comment]
Hey,

Thanks for the quick answer! Strangely, my project is still a boiler plate one. I rather feel that I miss something that you include. So far, I include tabulator.css, tabulator_simple.css, jquery (2.1.1 but I experimented with older and newer versions), jquery-UI 1.12.1 with only the core functions which I downloaded as you described on your page.

And, yes, I support your idea to keep Tabulator free from other libraries. It's not easy to find stuff now that's not infected with ideological platforms like angular.

Cheers,
Andreas

**Comment by olifolkerd** [comment]
I can see your problem there, you need to include tabulator.css, ***OR*** tabulator_simple.css not both.

I hope that clears things up.

Cheers

Oli

**Comment by fortrax** [comment]
Thanks for the hint, Oli. But even with only one of the two CSS files included, the unwanted behaviour remains. In the (unlikely) event that you might be interested to debug this on my computer, e.g. by Skype: my Skype name is andreas.fordan

Another remark: When experimenting with different versions, it might be helpful to have a header comment in your files containing the version number.

Cheers,
Andreas

**Comment by fortrax** [comment]
Maybe a relevant fact: When I use the mouse really fast moving a column header, then the mouse pointer is faster than the placeholderElement which lags a bit behind. The "mouseover" works for this fraction of a second and the column moves to that place (but not further). So the reason is something like IE finds the placeholderElement to be "not tranparent" enough to let the "mouseover" through.

**Comment by olifolkerd** [comment]
Which of the two css files did you use?

When you are dragging the column headers where is your mouse pointer? over the table, over the headers or outside the table? the same goes for when you release the click?

Could you post a copy of the table constructor you are testing with?

As it happens, I am looking to add source version numbers to the 33 release onwards :)

Cheers

Oli

**Comment by fortrax** [comment]
Either of the two css files does not work.

Of course, while dragging a column header one tries to hit other column headers. Anywhere else I would not really expect it to work and it appears so. As I said, in Chrome, the behaviour is absolutely as expected. After releasing (mouseup) the column snaps back.

Table constructor is still the one from the initial post.

Cheers & thanks,
Andreas

**Comment by olifolkerd** [comment]
Hmmm, that is strange,

I cant seem to replicate the issue, is it only IE 11 that is having issues for you?

**Comment by fortrax** [comment]
I did not test other browsers than IE11 and Chrome, sorry. I spent a bit time on it but I could not advance an inch. I tried to create the absolute minimal page possible. Note that I use ASP.NET MVC which generates the HTML from a "cshtml" format. But the html part is the following:

```
<!DOCTYPE html>
<html>
<head>
    <title>Document Overview</title>
    <link href="http://olifolkerd.github.io/tabulator/css/3.2/tabulator_site.css" rel="stylesheet">
    <script src="http://code.jquery.com/jquery-2.2.4.min.js" integrity="sha256-BbhdlvQf/xTY9gja0Dq3HiwQF8LaCRTXxZKRutelT44=" crossorigin="anonymous"></script>
    <script src="http://code.jquery.com/ui/1.12.1/jquery-ui.min.js" integrity="sha256-VazP97ZCwtekAsvgPBSUwPFKdrwD3unUfSGVYrahUqU=" crossorigin="anonymous"></script>

    <script src="http://olifolkerd.github.io/tabulator/js/3.2/tabulator.min.js?i=1"></script>

    <script type="text/javascript">
        $(document).ready(function () {
            $("#overviewtable").tabulator({
                movableColumns: true, columns: [
                    { title: "Name", field: "name", width: 200 },
                    { title: "Date Of Birth", field: "dob", sorter: "date" },
                ],
            });
        });
    </script>
</head>
<body>
    <div id="overviewtable"></div>
</body>
</html>

```

I am purposely referring to the same URLs that you refer to on your page. And it still shows the error! I can't believe that it should be ASP.NET MVC to make the difference?! Note that when I use tabulator.css rather than tabulator_site.css, then this looks like without any css. It is weird that you never see the problem and never got rid of it (except on your page).

**Comment by fortrax** [comment]
Wow - it is MVC! I created a non-MVC app and - yahoo - I can move the columns. So if you want to reproduce the problem, run Visual Studio (2015 in my case) and create an ASP.NET MVC application. I am absolutly clueless about the different behaviour of IE because from its perspective there should be no difference, especially no such subtle CSS behaviour difference.

**Comment by olifolkerd** [comment]
Well that is kind of good news!

Im afraid I am a PHP developer and do not run .NET so would have no way to test this myself.

As this appears to be the result of an interaction with the .NET framework, i will close the ticket, but if you get to the bottom of the strange interaction then i would love to know if there are any tweaks i can make to Tabulator to help you out.

Let me know if you have any other issues

Cheers

Oli :)

**Comment by fortrax** [comment]
Now I spent another few hours on that problem. Whenever I created a new project, it worked, but my legacy project did not work however I stripped it absolutely naked. What I had not considered was web.config. The line
`<add name="X-UA-Compatible" value="IE=10"`
makes IIS add some annotation in the response header which switches IE11 in a compatibility mode - that doesn't let mouseover events through your placeholder. You were absolutely right closing the thread and I apologize having bothered you!

Cheers,
Andreas
